File: swagger/models/client_document.py
from .base_model import BaseModel


class ClientDocument(BaseModel):
	"""
	no description available

	Attributes:

		file_name
		The name of the file to be used on the client’s documents page when
		it is uploaded.

		media_type
		The type of file or file extension. Possible values are:  * pdf  * jpg
		* jpeg  * tif  * tiff  * png  * doc  * docx  * bmp  * txt  * rtf  *
		xlsx  * xls  * gif  * zip  * ppt  * pptx  * mov

		buffer
		A Base64-encoded string representation of the file’s byte array.
	"""

	input_map = {
		'FileName': 'file_name',
		'MediaType': 'media_type',
		'Buffer': 'buffer',
		}

	output_map = {
		'file_name': 'FileName',
		'media_type': 'MediaType',
		'buffer': 'Buffer',
		}

	def __init__(self, file_name, media_type, buffer):
		self.file_name = file_name
		self.media_type = media_type
		self.buffer = buffer

File: swagger/Models/ClientDocument.php
<?php

namespace App\Mindbody\Models;

/**
 * no description available
 */
class ClientDocument extends BaseModel
{
	/**
	 * The name of the file to be used on the client’s documents page when
	 * it is uploaded.
	 * 
	 * @var string 
	 */
	public $fileName;
	/**
	 * The type of file or file extension. Possible values are:  * pdf  * jpg
	 * * jpeg  * tif  * tiff  * png  * doc  * docx  * bmp  * txt  * rtf  *
	 * xlsx  * xls  * gif  * zip  * ppt  * pptx  * mov
	 * 
	 * @var string 
	 */
	public $mediaType;
	/**
	 * A Base64-encoded string representation of the file’s byte array.
	 * 
	 * @var string format:byte
	 */
	public $buffer;
	/**
	 * Constructor
	 */
	public function __construct($fileName, $mediaType, $buffer)
	{
		$this->fileName = $fileName;
		$this->mediaType = $mediaType;
		$this->buffer = $buffer;
	}

	protected function getInputMap()
	{
		return [
			'FileName' => ['fileName', 'string', null],
			'MediaType' => ['mediaType', 'string', null],
			'Buffer' => ['buffer', 'string', 'byte'],
			];
	}

	protected function getOutputMap()
	{
		return [
			'fileName' => ['FileName', 'string', null],
			'mediaType' => ['MediaType', 'string', null],
			'buffer' => ['Buffer', 'string', 'byte'],
			];
	}
}
